### DISCUSSION TITLE
Where to configure exposeAllEnv?

### DISCUSSION THREAD

**Discussion by flheisig** [renovatebot/renovate#10667] [Request Help] [answered]
Hi,

we are trying to use renovate in a self-hosted configuration on Gitlab. The repository to be scanned contains a Gradle build file which expects the credentials for Artifactory as environment variables, which works fine in our build pipeline.

When running renovate we get error messages like `Gradle extraction failed`, `Command failed: ./gradlew --init-script renovate-plugin.gradle renovate`, `Could not get unknown property 'ARTIFACTORY_USER'`. So I suppose the environment variable is not available to the renovate process. 

After reading the docs I think that setting `exposeAllEnv=true` will solve my problem. I tried giving it as a command line flag (`renovate --exposeAllEnv=true`), which returns `error: unknown option '--exposeAllEnv=true'`. Setting it in the renovate.json of the runner does not seem to have any effect.

**Comment by rarkins** [comment, marked as answer]
Use `--expose-all-env=true` in CLI instead

**Comment by viceice** [comment]
Switch to `gradle-light` manager with  `:switchToGradleLite` preset, so no gradle execution is required. It will also a lot faster to extract deps and lower resources are used.

https://docs.renovatebot.com/presets-default/#switchtogradlelite
